tests/functional/nccl_message_transfer.c

```diff
@@ -30,35 +31,45 @@ int main(int argc, char* argv[])
 	/* Initialisation for data transfer */
 	nccl_net_ofi_req_t *req[NUM_REQUESTS] = {NULL};
 	void *mhandle[NUM_REQUESTS];
+	char handle[NCCL_NET_HANDLE_MAXSIZE];
```

Comment:
There are a bunch of changes in this commit that are reorganizing code and unrelated to the core purpose of this commit (changing the device id handling).  Please move those into their own commits (at least one for hoisting variables to the top of the function and another for moving iterator declarations into the for loop, possibly more but use your judgement).